src/HOI4World/Map/Railways.cpp

```diff
@@ -14,152 +16,221 @@ using HoI4::Railway;
 
-		std::priority_queue<std::vector<int>> newPaths;
-		newPaths.push({vic2ProvinceNum});
+		new_paths.push_back(new_path);
+	}
 
-		while (!newPaths.empty())
-		{
-			auto path = newPaths.top();
-			newPaths.pop();
+	return new_paths;
+}
 
-			const auto lastProvince = path[path.size() - 1];
-			for (const auto& neighborNumber: Vic2MapData.GetNeighbors(lastProvince))
-			{
-				if (handledProvinces.contains(neighborNumber))
-				{
-					continue;
-				}
-				handledProvinces.insert(neighborNumber);
 
-				const auto neighborItr = Vic2Provinces.find(neighborNumber);
-				if (neighborItr == Vic2Provinces.end())
-				{
-					continue;
-				}
-				const auto& neighborProvince = neighborItr->second;
-				if (!neighborProvince->isLandProvince())
-				{
-					continue;
-				}
+std::set<std::vector<int>> DetermineAllVic2PossiblePathsForProvince(int vic2_province_num,
+	 const std::set<int>& valid_vic2_province_numbers,
+	 const std::map<int, std::shared_ptr<Vic2::Province>>& vic2_provinces,
+	 const Maps::MapData& vic2_map_data)
+{
+	std::set handled_provinces{vic2_province_num};
+	std::set<std::vector<int>> potential_new_vic2_possible_paths;
 
-				auto newPath = path;
-				newPath.push_back(neighborNumber);
-				if (validVic2ProvinceNumbers.contains(neighborNumber))
-				{
-					potentialNewVic2ProvincePaths.insert(newPath);
-				}
-				else
-				{
-					newPaths.push(newPath);
-				}
-			}
-		}
+	std::priority_queue<std::vector<int>> new_paths;
+	new_paths.push({vic2_province_num});
+
+	while (!new_paths.empty())
+	{
+		std::vector<int> path = new_paths.top();
+		new_paths.pop();
 
-		std::vector<int> bestPotentialNewPath;
-		long potentialNewPathScore = 0;
-		for (const auto& potentialNewVic2ProvincePath: potentialNewVic2ProvincePaths)
+		for (const auto& new_path:
+			 GetNextPaths(path, valid_vic2_province_numbers, vic2_provinces, vic2_map_data, handled_provinces))
 		{
-			if (potentialNewVic2ProvincePath.size() == 2)
+			if (valid_vic2_province_numbers.contains(new_path.back()))
 			{
-				auto reversedPath = potentialNewVic2ProvincePath;
-				std::ranges::reverse(reversedPath);
-				if (!vic2ProvincePaths.contains(reversedPath))
-				{
-					vic2ProvincePaths.insert(potentialNewVic2ProvincePath);
-				}
-				continue;
+				potential_new_vic2_possible_paths.insert(new_path);
 			}
-
-			const int potentialLastProvinceNum = potentialNewVic2ProvincePath[potentialNewVic2ProvincePath.size() - 1];
-			const auto potentialLastProvinceItr = Vic2Provinces.find(potentialLastProvinceNum);
-			if (potentialLastProvinceItr == Vic2Provinces.end())
+			else
 			{
-				continue;
+				new_paths.push(new_path);
 			}
-			const int potentialLastProvincePopulation = potentialLastProvinceItr->second->getPopulation();
-			if (potentialLastProvincePopulation > potentialNewPathScore)
+		}
+	}
+
+	return potential_new_vic2_possible_paths;
+}
+
+
+int GetProvincePopulation(int province_num, const std::map<int, std::shared_ptr<Vic2::Province>>& vic2_provinces)
```

Comment:
I'm somewhat surprised population had anything to do with railways. Far from it being inconvenient or anything, but if I was a victorian-era rail-layer I'd pick those provinces that had flattest possible terrain. Hoi4 provinces don't have pops anyway so it wouldn't matter to hoi4 laborers.

Reply:
Oh gosh. This was old code from when I only took some of the potential Vic2 paths. I've gone and stripped it wholesale.